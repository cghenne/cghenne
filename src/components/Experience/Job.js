// @flow
import React from 'react';
import type { JobT } from './constants';
import './Experience.css';

type Props = JobT;

const Job = ({
  start_date,
  end_date,
  position,
  company,
  location,
  description,
  tech_stack
}: Props) => {
  return (
    <div className="job">
      <div className="dates">
        {start_date} - {end_date ? end_date : 'Present'}
      </div>
      <div className="details">
        <h4>{position}</h4>
        <h5>{`${company}, ${location}`}</h5>

        {description && <p>{description}</p>}
        {tech_stack && (
          <p>
            <span className="bold">Tech stack: </span>
            {tech_stack}
          </p>
        )}
      </div>
    </div>
  );
};

Job.displayName = 'Job';

export default Job;
