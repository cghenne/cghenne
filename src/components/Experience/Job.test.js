import React from 'react';
import { shallow } from 'enzyme';
import Job from './Job';

const fakeJob = {
  start_date: 'April 2018',
  end_date: null,
  position: 'Lead Front-End Engineer',
  company: 'NA',
  location: 'Toronto, CA',
  description: null,
  tech_stack: null
};

describe('Job', () => {
  const wrapper = shallow(<Job {...fakeJob} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the job in a job class', () => {
    expect(
      wrapper
        .find('div')
        .first()
        .prop('className')
    ).toEqual('job');
  });

  it('should render "start_date - Present" if the end_date is not defined', () => {
    expect(wrapper.find('.dates').text()).toEqual(
      `${fakeJob.start_date} - Present`
    );
  });

  it('should render "start_date - end_date" if the end_date is defined', () => {
    const end_date = 'June 2025';
    wrapper.setProps({ end_date });
    expect(wrapper.find('.dates').text()).toEqual(
      `${fakeJob.start_date} - ${end_date}`
    );
  });

  it('should render the details of the job in a details class', () => {
    const details = wrapper.find('.details');
    expect(details.exists()).toEqual(true);
    expect(details.find('h4').text()).toEqual(fakeJob.position);
    expect(
      details
        .find('h5')
        .first()
        .text()
    ).toEqual(fakeJob.company);
    expect(
      details
        .find('h5')
        .last()
        .text()
    ).toEqual(fakeJob.location);
  });

  it('should not render the description if the description is not defined', () => {
    expect(wrapper.find('.description').exists()).toEqual(false);
  });

  it('should render the description if the description is defined', () => {
    const description = 'Awesome Job Description';
    wrapper.setProps({ description });
    expect(wrapper.find('.description').text()).toEqual(description);
  });

  it('should not render the tech stack if the tech stack is not defined', () => {
    expect(wrapper.find('.tech-stack').exists()).toEqual(false);
  });

  it('should render the tech stack if the tech stack is defined', () => {
    const tech_stack = 'Awesome Tech Stack';
    wrapper.setProps({ tech_stack });
    expect(wrapper.find('.tech-stack').html()).toContain(tech_stack);
  });
});
