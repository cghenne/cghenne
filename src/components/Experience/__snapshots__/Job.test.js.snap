// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job should render without crashing 1`] = `
<div
  className="job"
>
  <div
    className="dates"
  >
    July 2019
     - 
    Present
  </div>
  <div
    className="details"
  >
    <h4>
      Senior Front-End Engineer
    </h4>
    <h5>
      Uber
       - 
      Toronto, CA
    </h5>
  </div>
</div>
`;
